.changeset/silver-pigs-care.md

```diff
@@ -2,11 +2,12 @@
 '@ag.ds-next/react': minor
 ---
 
-box: Add `focusAll` prop to enable a focus ring when the element is programmatically focused.
+box: Deprecate `focus` prop in favour of new `focusFor` prop to enable a focus ring when the element is keyboard focused (`focusFor="keyboard"`) and programmatically focused (`focusFor="all"`).
```

Comment:
Now that I've seen `focusFor` in use, do you think `focusRingFor` or `showFocusRing` are better alternatives?

Reply:
Maybe `focusRingFor`. Calling out the "focus ring" might be useful but no matter what you'll need to read the JSDoc to fully understand it.

Reply:
I wanted `programmatic` in there but it seems too long and obtuse 🤷

Reply:
Yeah, definitely need the JSDoc to fully understand, but I think adding "Ring" will make its intention much clearer when you see it used in your apps/components

Reply:
Yeah, ok, I'll migrate to `focusRingFor`.